### PR TITLE
Downgrade go pg

### PR DESCRIPTION
**Pull request by srieckhof**
Detailed changes description:

1. revert all changes that are due to changes in go-pg from this previous commit https://github.com/pace/bricks/commit/a4d7cc1303122875e4fb3efaf30387a44917a342#diff-96a9c4960a91cf3bb9fd115d50c8e7ef0f98de9cb856e9d4c09465d958a4cf0c
2. golang-ci.yml
- update go version from 1.21.3 to 1.21.12 because this fixes vulnarability issues
- install docker-compose before running the integration tests
- actually run the integration and not (again) the unit tests as was the case with "make ci"
- add "make lint" step to CI
3. fix issues resulting from "make lint"
4. fix integration test in cache.go

closes https://github.com/pace/bricks/issues/373

### CODE REVIEW

File: backend/postgres/postgres.go [hunk 1]
```diff
@@ -6,6 +6,10 @@ package postgres
 import (
 	"context"
 	"fmt"
+	"github.com/opentracing/opentracing-go"
```
Comment (arnold-iakab) [inline backend/postgres/postgres.go:9]: Please order imports accordingly

File: http/jsonapi/response_test.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"github.com/stretchr/testify/require"
```
Comment (arnold-iakab) [inline http/jsonapi/response_test.go:10]: Please order imports accordingly

File: backend/postgres/postgres.go (hunk 1 above)
Comment (srieckhof) [reply backend/postgres/postgres.go]: Sure. See following commit with `$ gofumpt -w .` having been executed on the whole project.

File: http/jsonapi/response_test.go (hunk 2 above)
Comment (srieckhof) [reply http/jsonapi/response_test.go]: Same as above.

**Review by arnold-iakab — APPROVED** [review]
lgtm

**Pull request merged**